client/jetpack-cloud/sections/pricing/header/style.scss

```diff
@@ -1,6 +1,10 @@
 @import '@wordpress/base-styles/breakpoints';
 @import '@wordpress/base-styles/mixins';
 
+#header {
+	display: none;
+}
```

Comment:
This CSS rule says that the `#header` element should _always_ be hidden. It's not scoped to any section. And as soon as the stylesheet is loaded, it will indeed hide it. And it will remain hidden even if we navigate away from the Pricing section to another route, even one that wants to show the masterbar.

Reply:
The Jetpack Cloud Pricing handler already disables the masterbar here:

https://github.com/Automattic/wp-calypso/blob/trunk/client/jetpack-cloud/sections/pricing/controller.tsx#L32

Do you know why it doesn't work?

Reply:
> And as soon as the stylesheet is loaded, it will indeed hide it. And it will remain hidden even if we navigate away from the Pricing section to another route, even one that wants to show the masterbar.

Oh, thanks. Didn't realize that but it does show up when you navigate away from pricing page. 🤔 🤷 

> Do you know why it doesn't work?

Please see the [comment](https://github.com/Automattic/wp-calypso/pull/59806#issuecomment-1007181077) above. That masterbar is disabled pretty late and on a slow connection, users may see that bar until the page loads.

Reply:
Pushed a fix to restrict the CSS rule to pricing section. Thanks

Reply:
This is better now, but I'd still like to understand better why other solutions don't work. First, I can't reproduce the issue even on a Slow 3G connection. The masterbar never renders on `/pricing` load. Maybe it does on a navigation from/to some other page? Do you have a video of the bug?

Also, the solution that adds `jetpack-cloud-pricing` to `noMasterbarForSection` should also work. The `.is-section-jetpack-cloud-pricing` body CSS class is added by the `Layout` component literally at the same time, during the same render, when the `jetpack-cloud-pricing` section is being activated. And it does that in a way that's established and used by other parts of Calypso.

Another suspicious thing I found that might be related is [this `showMasterbar` effect](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/jetpack-plans/selector.tsx#L112) in the plans selector component, added by @monsieur-z in #59324. Is it possible that the `SelectorPage` component unmounts while the `/pricing` route is still active? An individual component somewhere down the tree shouldn't do `hideMasterbar` or `showMasterbar`. Only a route handler or the `Layout` component should do that because it's a responsibility of the top-level page layout.

To summarize, this patch is mergeable and does the job (or at least doesn't do harm) but I believe there should be a better solution and would like to understand the problem better.

Reply:
Steps for devs (See the GIF attached):

- Goto https://cloud.jetpack.com/pricing
- Open Dev Tools > Redux
- In filter input, enter `MASTERBAR_TOGGLE_VISIBILITY`
- Hover over to the action item below and click on **Jump**
- Click on the **Forward** and **Backward** buttons to toggle the visibility of the masterbar


DEMO
![demo](https://user-images.githubusercontent.com/18226415/148535499-f0de466f-63a8-4704-bc31-ddc57f329172.gif)

Reply:
> To summarize, this patch is mergeable and does the job (or at least doesn't do harm) but I believe there should be a better solution and would like to understand the problem better.

Yeah, I agree. But I am new here and not a master of calypso routing yet. 😄

Reply:
> The `.is-section-jetpack-cloud-pricing` body CSS class is added by the `Layout` component literally at the same time, during the same render, when the `jetpack-cloud-pricing` section is being activated.

Also, I see that `is-section-jetpack-cloud-pricing` class is added before anything is displayed on the page.

Reply:
> Steps for devs (See the GIF attached):

That's an artificial scenario though, that does time-travel in the Redux devtool. Can I reproduce the bug by simply loading the `/pricing` page and observing it?

Before this patch, there should appear an empty page with a Jetpack masterbar, and then the pricing page without the masterbar, is that right? There was never a "pricing page with masterbar" combination, right?

When I'm trying to reproduce this, I only ever see an "empty page", never an "empty page with masterbar". I know that such a page is rendered by React at least once, and Redux time-travel shows that, but it's never painted because it's immediately (within the same event loop tick) superseded by an updated render.

> not a master of calypso routing yet.

Don't worry about it, nobody is a master of that 🙂

Reply:
> Can I reproduce the bug by simply loading the `/pricing` page and observing it?

Yes, it was reported here p8oabR-LV-p2#comment-5760 and I can confirm it using an ultra slow custom throttle in my browser.

Reply:
I haven't verified this yet, but could it be that users who have access to the Partners Portal are seeing that?

Reply:
> could it be that users who have access to the Partners Portal are seeing that?

No, it's shown regardless.

Reply:
Ok. I'm seeing it now. One has to start from a site. In other words, there must be a site in the URL (i.e., the path must look like `/pricing/:site`).

Reply:
I suspect that having a site in the URL makes the app dispatch some requests, which delay the entire rendering process, and one is able to see the masterbar for a brief moment.

Reply:
Yeah I noticed that Eric's P2 report uses a `/pricing/:site` URL, not just `/pricing`.